test/extensions/filters/http/proto_api_scrubber/filter_config_test.cc

```diff
@@ -0,0 +1,511 @@
+TEST_F(ProtoApiScrubberFilterConfigTest, MatchTreeValidation) {
+  absl::StatusOr<std::shared_ptr<const ProtoApiScrubberFilterConfig>> filter_config =
+      ProtoApiScrubberFilterConfig::create(proto_config_, factory_context_);
+  ASSERT_EQ(filter_config.status().code(), absl::StatusCode::kOk);
+  ASSERT_NE(filter_config.value(), nullptr);
+  filter_config_ = filter_config.value();
+
+  NiceMock<StreamInfo::MockStreamInfo> mock_stream_info;
+  Http::Matching::HttpMatchingDataImpl http_matching_data_impl(mock_stream_info);
+
+  // Testing the match_tree for the field_mask `debug_info`.
+  // The match expression in hardcoded to `true` for `debug_info` which should result in a match
+  // and a corresponding action named `RemoveFieldAction`.
+  MatchTreeHttpMatchingDataSharedPtr match_tree =
+      filter_config_->getRequestFieldMatcher("/library.BookService/GetBook", "debug_info");
+  ASSERT_NE(match_tree, nullptr);
+  Matcher::MatchTree<HttpMatchingData>::MatchResult match_result =
+      match_tree->match(http_matching_data_impl);
+  ASSERT_EQ(match_result.match_state_, Matcher::MatchState::MatchComplete);
+  ASSERT_TRUE(match_result.on_match_.has_value());
+  EXPECT_EQ(match_result.on_match_.value().action_cb_()->typeUrl(),
+            "envoy.extensions.filters.http.proto_api_scrubber.v3.RemoveFieldAction");
+
+  // Testing the match_tree for the field_mask `book.debug_info`.
+  // The match expression in hardcoded to `false` for `book.debug_info` which should result in
+  // no match.
+  match_tree =
+      filter_config_->getResponseFieldMatcher("/library.BookService/GetBook", "book.debug_info");
+  ASSERT_NE(match_tree, nullptr);
+  Matcher::MatchTree<HttpMatchingData>::MatchResult match_result2 =
+      match_tree->match(http_matching_data_impl);
+  ASSERT_EQ(match_result2.match_state_, Matcher::MatchState::MatchComplete);
+  EXPECT_FALSE(match_result2.on_match_.has_value());
+
+  // Validate invalid method name for request and response field matchers.
+  match_tree = filter_config_->getRequestFieldMatcher("/non.existent.service/method", "book");
+  ASSERT_EQ(match_tree, nullptr);
+  match_tree =
+      filter_config_->getResponseFieldMatcher("/non.existent.service/method", "book.debug_info");
+  ASSERT_EQ(match_tree, nullptr);
+
+  // Validate invalid field mask for request and response field matchers.
+  match_tree = filter_config_->getRequestFieldMatcher("/library.BookService/GetBook",
+                                                      "non.existent.field.mask");
+  match_tree = filter_config_->getResponseFieldMatcher("/library.BookService/GetBook",
+                                                       "non.existent.field.mask");
+  ASSERT_EQ(match_tree, nullptr);
+}
+
+TEST_F(ProtoApiScrubberFilterConfigTest, MethodNameValidations) {
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context;
+  absl::StatusOr<std::shared_ptr<const ProtoApiScrubberFilterConfig>> filter_config;
+
+  filter_config =
```

Comment:
nit: surround each sub-test in its own scope ({ ... }) to emphasis that the tests are disjoint.